File: 21 Dec Count X in Range of a Sorted Array/code.cpp
// You are given a sorted array arr[] and a 2D array queries[][], where queries[i] represents a query in the form [l, r, x]. For each query, count how many times the number x appears in the subarray arr[l...r] (inclusive).

// ans:
// just ub and lb things

class Solution {
  public:
    vector<int> countXInRange(vector<int>& arr, vector<vector<int>>& q) {
        // code here
        vector<int>res;
        int n=arr.size();
        for(auto c:q){
            
            res.push_back(upper_bound(begin(arr)+c[0],begin(arr)+c[1]+1,
            c[2])-lower_bound(begin(arr)+c[0],begin(arr)+c[1]+1,c[2]));
        }
        return res;
    }
};

File: 8 Jan Count Subarrays with k odds/code.py
class Solution:
    def countSubarrays(self, arr, k):
        n=len(arr)
        def rec(o):
            i,j=0,0
            res,c=0,0
            while(i<n):
                c+=(arr[i]%2)
                while(j<=i and c==o):
                    res+=(n-i)
                    c-=(arr[j]%2)
                    j+=1
                i+=1
            return res
        return rec(k)-rec(k+1)